src/lib/TopLevel.hs

```diff
@@ -4,60 +4,41 @@
 -- license that can be found in the LICENSE file or at
 -- https://developers.google.com/open-source/licenses/bsd
 
-{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE FlexibleContexts #-}
-{-# LANGUAGE OverloadedStrings #-}
 
 module TopLevel (evalSourceBlock, evalDecl, evalSource, evalFile,
-                 EvalConfig (..), Backend (..), initializeBackend,
-                 substArrayLiterals) where
+                 EvalConfig (..), Backend (..)) where
 
-import Control.Concurrent.MVar
 import Control.Monad.State.Strict
 import Control.Monad.Reader
 import Control.Monad.Except hiding (Except)
 import Data.Text.Prettyprint.Doc
-import Foreign.Ptr
 import Data.List (partition)
-import Data.Maybe (fromJust, fromMaybe)
 import Data.Time.Clock (getCurrentTime, diffUTCTime)
 
-import Array
 import Syntax
 import Cat
 import Env
 import Embed
 import Type
 import Inference
+import Interpreter
 import Simplify
 import Serialize
 import Imp
-import Interpreter
 import JIT
 import Logging
 import LLVMExec
 import PPrint
 import Parser
 import Util (highlightRegion)
 import Optimize
-import CUDA
 
-data Backend = LLVM | LLVMCUDA | LLVMMC | Interp | JAX  deriving (Show, Eq)
 data EvalConfig = EvalConfig
-  { logFile     :: Maybe FilePath
-  , evalEngine  :: BackendEngine
-  , logService  :: Logger [Output] }  -- TODO: Move this to TopPassM
-
-data LLVMEngineKind = Serial | Multicore | CUDA
-data BackendEngine = LLVMEngine LLVMEngineKind LLVMEngine
-                   -- | JaxServer JaxServer
-                   | InterpEngine
-
-type LLVMEngine = MVar (Env (Ptr ()))
--- type JaxServer = PipeServer ( (JaxFunction, [JVar]) -> ([JVar], String)
---                            ,( [JVar] -> [Array]
---                            ,( Array -> ()  -- for debugging
---                            ,())))
+  { backendName :: Backend
+  , preludeFile :: FilePath
```

Comment:
I don't see why would we want to put `preludeFile` here. `EvalConfig` is a really bad name for this struct in general. It's what we use to keep the state between different invocations, and the prelude path is definitely not part of this state (+ there might not even be a prelude file, once you revert the revert of #241).